src/react-integration/__tests__/subscriptions.tsx

```diff
@@ -3,110 +3,72 @@ import { cleanup, renderHook, act } from 'react-hooks-testing-library';
 
 import nock from 'nock';
 
-import {
-  PollingArticleResource,
-} from '../../__tests__/common';
+import { PollingArticleResource } from '../../__tests__/common';
 import { useSubscription, useCache } from '../hooks';
-import RestProvider from '../provider';
-import NetworkManager from '../../state/NetworkManager';
-import SubscriptionManager from '../../state/SubscriptionManager';
-import PollingSubscription from '../../state/PollingSubscription';
-import { FetchAction, ReceiveAction } from '../../types';
-
-class MockNetworkManager extends NetworkManager {
-  handleFetch(action: FetchAction, dispatch: React.Dispatch<any>) {
-    const mockDispatch: typeof dispatch = (v: any) => {
-      act(() => {
-        dispatch(v);
-      });
-    };
-    return super.handleFetch(action, mockDispatch);
-  }
-  handleReceive(action: ReceiveAction) {
-    act(() => {
-      super.handleReceive(action);
-    });
-  }
-}
-class MockPollingSubscription extends PollingSubscription {
-  update() {
-    act(() => {
-      super.update();
-    });
-  }
-}
+import createRenderRestHook from '../../test/helper';
+import {
+  makeRestProvider,
+  makeExternalCacheProvider,
+} from '../../test/providers';
 
 afterEach(cleanup);
 
-describe('<RestProvider /> with subscriptions', () => {
-  const articlePayload = {
-    id: 5,
-    title: 'hi ho',
-    content: 'whatever',
-    tags: ['a', 'best', 'react'],
-  };
+for (const makeProvider of [makeRestProvider, makeExternalCacheProvider]) {
```

Comment:
this tabbed over most of the file making it look at way more changed. truthfully only the use of createRenderRestHook was changed